compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -1870,6 +1894,23 @@ public void visit(BLangWaitForAllExpr waitForAllExpr) {
         }
     }
 
+    private BRecordType getWaitForAllExprReturnType(List<BLangWaitForAllExpr.BLangWaitKeyValue> keyVals) {
+        BRecordType retType = new BRecordType(null);
+        
+        for (BLangWaitForAllExpr.BLangWaitKeyValue keyVal : keyVals) {
+            BSymbol symbol = symResolver.lookupSymbol(env, names.fromIdNode(keyVal.key), SymTag.VARIABLE);
```

Comment:
I think what we expect here is to get the constraint type of the `future`
If so I think we need to do something in the line of 
```java
BType t = ((BFutureType) checkExpr(keyVal.valueExpr, env)).constraint
```

Reply:
Not quite. Consider something like below:
```ballerina
int x = wait {w1, w2};
```
What I want is to create a record type for the RHS so that in the error message I can say expected `int`, but found this type of record. If I call checkExpr(), it'll go in the normal type checking flow, which means if there are any errors there, the compiler will log errors for those as well.

Reply:
If I understand correctly your concern is around duplicate error message due to `checkExpr` right,
```ballerina
 int x = wait { w1+1, nonFutureRef };
```
Because of caching I don't think there will be multiple error messages.

Anyway for `wait { someKey: w1, someKey1: w1 }` resolving symbol for the `key` will always result in undefined symbol (other)

Reply:
Fixed this with 1be935a. Have a look